test/open-report.js

```diff
@@ -0,0 +1,77 @@
+import test from 'ava';
+import proxyquire from 'proxyquire';
+
+const openReport = proxyquire('../lib/open-report', {
+	'open-editor': files => files
+});
+
+const createReport = ({results = [], errorCount = 0, warningCount = 0} = {}) => ({
+	results, errorCount, warningCount
+});
+
+const createResult = ({errorCount = 0, warningCount = 0, filePath = '', messages = []} = {}) => ({
+	errorCount, warningCount, filePath, messages
+});
+
+const createMessage = ({line = 1, column = 10} = {}) => ({
+	line, column
+});
+
+test('opens nothing when there are no errors nor warnings', t => {
+	const expected = openReport(createReport());
+	const actual = [];
+
+	t.deepEqual(actual, expected);
+});
+
+test('only opens errors if there are errors and warnings', t => {
+	const report = createReport({
+		errorCount: 1,
+		warningCount: 2,
+		results: [
+			createResult({errorCount: 7, filePath: 'seven-errors.js', messages: [createMessage({line: 7, column: 17})]}),
+			createResult({warningCount: 3, filePath: 'three-warnings.js', messages: [createMessage({line: 3, column: 13})]}),
+			createResult({errorCount: 1, filePath: 'one-error.js', messages: [createMessage()]}),
+			createResult({warningCount: 1, filePath: 'one-warning.js', messages: [createMessage()]})
+		]
+	});
```

Comment:
IMO, manually creating reports is rather fragile: when ESLint changes the report format, we have to change all those create-functions.

We are already testing on a higher level: we want to assert that the _output_ is correct (i.e. that `open-editor` is called with the right arguments), hence testing with _real input_ is sensible, for example:

```js
const xo = require('..');
// …
const report = xo.lintText(`// TODO xxx
const x =       4;
`, {filename: 'one-warning-and-two-errors.js'});
```

We could also reuse or add files to `test/fixtures` for more complicated tests.